QuantConnect.Polygon/PolygonSubscriptionManager.cs

```diff
@@ -215,7 +215,7 @@ private void ConnectWebSocket(PolygonWebSocketClientWrapper webSocket)
 
             if (result == 0)
             {
-                throw new PolygonAuthenticationException("Failed to authenticate websocket. Make sure your API key is valid and has the right permissions.");
+                throw new PolygonAuthenticationException($"WebSocket authentication failed for security types: {string.Join(", ", webSocket.SupportedSecurityTypes)}. Make sure your API key is valid and has the right permissions.");
```

Comment:
If we raise an exception, the user might lose information. For example, if they don't have Equity and Index, they will see the exception with Equity and miss Index. They will purchase Equity, redeploy, and get a message about Index. They will be mad we didn't tell them all subscription required. 

Do we need to throw? QuantConnect data provider doesn't throw to say it doesn't have options data.

Reply:
😎